ujson/shared/src/main/scala/diffson/ujson/package.scala

```diff
@@ -0,0 +1,191 @@
+      json.objOpt.map(_.toMap)
+
+    def makeArray(values: Vector[Value]): Value =
+      Arr.from(values)
+
+    def makeObject(fields: Map[String, Value]): Value =
+      Obj.from(fields)
+
+    def show(json: Value): String =
+      json.render()
+
+    def eqv(json1: Value, json2: Value): Boolean = json1 == json2
+  }
+
+  implicit val pointerEncoder: Writer[Pointer] =
+    implicitly[Writer[String]].comap(a => a.show)
+
+  implicit val pointerDecoder: Reader[Pointer] =
+    implicitly[Reader[String]].map { str =>
+      import cats.implicits._
+      Pointer.parse[Try](str).get
+    }
+
+  private val operationAsJson: Operation[Value] => Value = {
+    case Add(path, value) =>
+      Obj(
+        "op" -> Str("add"),
+        "path" -> Str(path.show),
+        "value" -> value,
+      )
+    case Remove(path, Some(old)) =>
+      Obj(
+        "op" -> Str("remove"),
+        "path" -> Str(path.show),
+        "old" -> old,
+      )
+    case Remove(path, None) =>
+      Obj(
+        "op" -> Str("remove"),
+        "path" -> Str(path.show),
+      )
+    case Replace(path, value, Some(old)) =>
+      Obj(
+        "op" -> Str("replace"),
+        "path" -> Str(path.show),
+        "value" -> value,
+        "old" -> old,
+      )
+    case Replace(path, value, None) =>
+      Obj(
+        "op" -> Str("replace"),
+        "path" -> Str(path.show),
+        "value" -> value,
+      )
+    case Move(from, path) =>
+      Obj(
+        "op" -> Str("move"),
+        "from" -> Str(from.show),
+        "path" -> Str(path.show),
+      )
+    case Copy(from, path) =>
+      Obj(
+        "op" -> Str("copy"),
+        "from" -> Str(from.show),
+        "path" -> Str(path.show),
+      )
+    case Test(path, value) =>
+      Obj(
+        "op" -> Str("test"),
+        "path" -> Str(path.show),
+        "value" -> value,
+      )
+  }
+
+  implicit val operationEncoder: Writer[Operation[Value]] =
+    implicitly[Writer[Value]].comap(operationAsJson)
+
+  def throwFieldExpected(
+    fieldName: String
+  ): PartialFunction[Throwable, Nothing] = {
+    case t: Throwable =>
+      throw new Exception(s"Expected field `$fieldName`, but it is missing", t)
+  }
+
+  private def decodeOperation(value: Value): Operation[Value] = {
+    def readPointer(value: Value, path: String = "path"): Pointer = {
+      val p = try { value.obj(path) } catch throwFieldExpected(path)
+      read[Pointer](p)
+    }
+
+    val op = try value.obj("op").str catch throwFieldExpected("op")
+
+    op match {
+      case "add" =>
+        val path = readPointer(value)
+        val v =
+          try value.obj("value")
+          catch throwFieldExpected("value")
+        Add(path, v)
+      case "remove" =>
+        val path = readPointer(value)
+        val old = value.objOpt.flatMap(_.get("old"))
+        Remove(path, old)
+      case "replace" =>
+        val path = readPointer(value)
+        val payload =
+          try value.obj("value")
```

Comment:
Isn't there a way with ujson to check for existence before failing instead of try/catch?

Reply:
Sure, there's the `.get(key)` method that behaves pretty much like in maps.

But we'll end up throwing exceptions anyway, it's the idiomatic way to propagate error in ujson/upickle.

Reply:
Rewrote the bridge without try/catch.

Reply:
I am not familiar with `ujson` but if that's the idiomatic way, then it's ok.